docs/index.html

```diff
@@ -0,0 +1,199 @@
+<html>
+<head>
+    <link rel="stylesheet" href="main.css">
+    <meta charset="UTF-8">
+    <title>Ambianic Serverless functions API Definition</title>
+                                                         </head>
+<body>
+<div class="container">
+    <h1>Ambianic Serverless functions API Definition</h1>
+    <p class="sw-info">
+        
+        Version: <span class="sw-info-version">0.1</span>
+    </p>
+    <p><p>An API Definition for Ambianic Serverless functions created using Swagger</p>
+    </p>
+    
+    
+    <div id="sw-schemes" class="sw-default-value">
+        <span class="sw-default-value-header">Schemes:</span>
+        
+    </div>
+    
+        <h2 id="swagger--summary-no-tags">Summary</h2>
+        <table class="table table-bordered table-condensed swagger--summary">
+            <thead>
+            <tr>
+                <th>Path</th>
+                <th>Operation</th>
+                <th>Description</th>
+            </tr>
+            </thead>
+            <tbody>
+                    <tr>
+                            <td class="swagger--summary-path" rowspan="1">
+                                <a href="#path--.netlify-functions-customers">/.netlify/functions/customers</a>
+                            </td>
+                        <td>
+                            <a href="#operation--.netlify-functions-customers-get">GET</a>
+                        </td>
+                        <td>
+                            
+                        </td>
+                    </tr>
+                    <tr>
+                            <td class="swagger--summary-path" rowspan="1">
+                                <a href="#path--subscriber">/subscriber</a>
+                            </td>
+                        <td>
+                            <a href="#operation--subscriber-post">POST</a>
+                        </td>
+                        <td>
+                            
+                        </td>
+                    </tr>
+                    <tr>
+                            <td class="swagger--summary-path" rowspan="1">
+                                <a href="#path--subscription">/subscription</a>
+                            </td>
+                        <td>
+                            <a href="#operation--subscription-delete">DELETE</a>
+                        </td>
+                        <td>
+                            
+                        </td>
+                    </tr>
+            </tbody>
+        </table>
+    
+    
+    
+    <h2>Paths</h2>
+    
+        
+        <span id="path--.netlify-functions-customers"></span>
+            <div id="operation--.netlify-functions-customers-get" class="swagger--panel-operation-get panel">
+                <div class="panel-heading">
+                    <div class="operation-summary"></div>
+                    <h3 class="panel-title"><span class="operation-name">GET</span> <strong>/.netlify/functions/customers</strong></h3>
```

Comment:
It's actually a problem originating from Stripe's `fetchcustomer` endpoint. It ideally should contain a means of getting a single user by email or any other recognizable field.

Rather It requires a `customer-id` which becomes a problem for us, since a database is not being used to store user's details

Reply:
This doesn't make any sense. `customer-id` becomes available as soon as the user logs in to their account in the PWA.

Reply:
Nope. Stripe generates a customers-id for each subscribed user. How do we preserve it for this usecase?

Take for example, a scenario where a user switches device.

What `customer-id` are you making reference to?

Reply:
My mistake, I mixed up Auth0 `user-id` with stripe's `customer-id`. 

Let's enforce a 1:1 mapping between Auth0 `user-id` and a Stripe's `customer-id` for now. The Auth0 user profile can store an attribute `stripe-customer-id` with the matching value.

In a separate comment I suggested simplifying the flow by implementing a cloud function which Stripe calls back when a subscription payment has been made. The function will take `customer-id` and `subscription-details` parameter. It will then updates the Auth0 user profile by setting a `premiumSubscriptionExpirationDate` attribute to the value provided by Stripe to Ambianic callback webhook function. From that point on the ambianic premium notification cloud functions will only have to verify if that `premiumSubscriptionExpirationDate` attribute exists and points to a date in the future before sending an email to the user via mailgun. If the attribute does not exists or points to an expired date, the user needs to be notified via administrative email and a PWA pop-up (the next time they open the PWA) that they will not receive premium notifications until their subscription is activated.


We will limit separately how many active devices a user can have at a time. Let's say up to 10 for now. Let's track this in a separate issue in this repo. The Auth0 Clients API and Devices API can be used to keep track and limit the number of authorized devices under a `user-id` account. Each device has a unique `client-id` obtained during the initial authorization flow.
https://auth0.com/docs/api/management/v2#!/Clients/get_clients
https://auth0.com/docs/api/management/v2#!/Device_Credentials/get_device_credentials

Reply:
The security issue has been addressed with storing the user’s stripe details in Auth0 profile.